CONTRIBUTING.md

```diff
@@ -66,43 +67,89 @@ Thank you for your contributions!!
 
 Join us in our [GitHub discussion board](https://github.com/aimakerspace/PeekingDuck/discussions) or [AI Singapore's Community forum](https://community.aisingapore.org/groups/computer-vision/forum/).
 
-. Post your thread regarding
+Post your thread regarding
 
 - Questions on the project
 - Suggestions/feature requests
 - Your custom nodes and projects
 
-## Code Styles
+## Project Conventions
 
 Help us maintain the quality of our project by following the conventions we take below.
 
-### Git Commit Messages
+### Branch Names
+
+It is a good practice to have a consistent branch naming approach. We follow the convention below, where words
+are separated by dashes (`-`).
+```bash
+<type>-<short description of task>
+  │
+  └─⫸ Commit Type: cicd|docs|feat|fix|node|refactor|test
+
+Examples:
+fix-linux-threading, node-crowd-counting, refactor-use-pathlib
+```
 
-This is a shortened version inspired by [angular's contributing docs](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-header).
+`<type>` must be one of the following:
+- cicd: Changes to CI/CD 
+- docs: Documentation changes
+- feat: Enhancements which are not new nodes
+- fix: Bug fixes
+- node: New nodes
+- refactor: Code changes that neither fixes a bug nor adds a feature
+- test: Unit or system tests
 
-A standard git commit message helps us structure the project. 
+
+### Git Commit Messages
+
+A standard git commit message makes it easier to read commit histories. This is a shortened version inspired by [angular's contributing docs](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-header).
 
 ```bash
-<type>(<scope>): <short summary>
-  │       │             │
-  │       │             └─⫸ Summary in present tense. Not capitalized. No period at the end.
-  │       │
-  │       └─⫸ Commit Scope: core|cli|changelog|packaging|models|heuristics|cicd-infra|docs-infra|community
+<type>: <short summary>
+  │            │
+  │            └─⫸ Summary in present tense. Not capitalized. No period at the end.
   │
-  │
-  └─⫸ Commit Type: build|ci|docs|feat|fix|perf|refactor|test
+  └─⫸ Commit Type: cicd|docs|feat|fix|node|refactor|test
 ```
+`<type>` must be one of the options described in the above [Branch Names](#branch-names) section.
+
+### Code Styles
+
+We follow the PEP8 style guide, with PEP484 type hinting for functions and methods, and use [Black](https://black.readthedocs.io/en/stable/) code formatter to ensure consistent code format.
+
+The following are commonly used conventions which we adhere to:
+- Imports
+  - Absolute imports instead of relative imports (e.g. avoid `from ...nodes.node import AbstractNode`)
+  - Use [isort](https://github.com/PyCQA/isort) to order imports
```

Comment:
`isort` requires some additional settings to play nice with `black` and to recognize `peekingduck` as first-party import.

The following section
```
[tool.isort]
profile = "black"
```
needs to be added to `pyproject.toml`.

Or this point can be removed if we do not play on enforcing using `isort` to sort imports.

Reply:
Let's just remove this point then